tests/pjit_test.py

```diff
@@ -4051,6 +4051,15 @@ def model(params, x):
 
     jax.vmap(jax.grad(model), in_axes=(None, 0))(params, x)  # doesn't crash
 
+  def test_jit_token_input(self):
+    x = jnp.arange(8)
+    token = jax.lax.create_token(None)
+    device = jax.devices()[0]
+    x = jax.device_put(x, device=device)
+    out1, out2 = jax.jit(lambda x, t: (x, t))(x, token)
+    self.assertArraysEqual(out1, x)
+    self.assertIsInstance(out2, core.Token)
```

Comment:
@yashk2810 Are you sure this test is meaningful? On my machine this runs fine even before the fix. It seems that only custom primitives trigger the code path that was previously broken.

Reply:
It failed on my machine with the same error you were hitting. After my fix, do you still see the error in your custom primitive?

Your impl rule will go through the jit path in the test above so it seems fine to me.

Reply:
You're right, I used the wrong JAX version for testing. Sorry for the noise.